docs/command_reference.md

```diff
@@ -0,0 +1,279 @@
+
+You can manage the changes that are staged by running:
+
+```bash
+# Check what changes are staged
+$ notary status <GUN>
+
+# Unstage a specific change
+$ notary status <GUN> --unstage 0
+
+# Alternatively, unstage all changes
+$ notary status <GUN> --reset
+```
+
+Note that `<GUN>` can take on arbitrary values, but when working with Docker Content Trust they are structured like `<url>/<account>/<repository>`.  For example `docker.io/library/alpine` for the [Docker Library Alpine image](https://hub.docker.com/r/library/alpine).
+
+When you're ready to publish your changes to the Notary server, run:
+
+```bash
+$ notary publish <GUN>
+```
+
+## Auto-publish changes
+
+Instead of manually running `notary publish` after each command, you can use the `-p` flag to auto-publish the changes from that command.
+For example:
+
+```bash
+$ notary init -p <GUN>
+```
+
+
+## Adding and removing trust data
+
+Users can sign content into a notary trusted collection by running:
+```bash
+$ notary add -p <GUN> <target_name> <target_file>
+```
+
+In the above command, the `<target_name>` corresponds to the name we want to associate the `<target_file>`'s signature with. Notary will sign the hash of the `<target_file>` into its trusted collection.
+Instead of adding a target by file, you can specify a hash directly:
+```bash
+$ notary addhash -p <GUN> <target_name> -sha256 <sha256Hash>
+```
+
+To check that your trust data was published successfully to the notary server, you can run:
+```bash
+$ notary list <GUN>
+```
+
+To remove targets from a trusted collection, you can run:
+```bash
+$ notary remove -p <GUN> <target_name>
+```
+
+## Delete trust data
+
+Users can remove all notary signed data for a trusted collection by running:
+
+```bash
+$ notary delete <GUN> --remote
+```
+
+If you don't include the `--remote` flag, Notary deletes local cached content
+but will not delete data from the Notary server.
+
+## Change the passphrase for a key
+
+The Notary CLI client manages the keys used to sign the trusted collection. These keys are encrypted at rest.
+To list all the keys managed by the Notary CLI client, run:
+
+```bash
+$ notary key list
+```
+
+To change the passphrase used to encrypt one of the keys, run:
+
+```bash
+$ notary key passwd <key_id>
+```
+
+## Rotate keys
+
+If one of the private keys is compromised you can rotate that key, so that
+content that was signed with those keys stop being trusted.
+
+For keys that are kept offline and managed by the Notary CLI client, such the
+keys with the root, targets, and snapshot roles, you can rotate them with:
+
+```bash
+$ notary key rotate <GUN> <key_role>
+```
+
+The Notary CLI client generates a new key for the role you specified, and
+prompts you for a passphrase to encrypt it.
+Then you're prompted for the passphrase for the key you're rotating, and if it
+is correct, the Notary CLI client contacts the Notary server to publish the
```

Comment:
Non-blocking:  wondering if we should call out that Notary CLI auto-publishes immediately, no flag needed?